jobrunner/run.py

```diff
@@ -150,6 +153,29 @@ def handle_single_job(job, api):
         raise
 
 
+def trace_handle_job(job, api, mode, paused):
```

Comment:
Just a thought, but would this be better written as a decorator that could be applied to `handle_job`?

I've been thinking that the tracer code is normally a good candidate for a decorator, but it's the first time where I've thought it could be nicely applied.

Reply:
The thing is, it's specific to the handle_job functionality. So a decorator would not be reusable, in which case, a function is similar.

The only reason its a function is a) we have two call sites, due to the synchronous logic and b) adding it handle_job would have meant indenting everything in a large function to be in a with block, which would have been a bad diff and just a bit awkward.